extensions/amp-form/0.1/amp-form.js

```diff
@@ -109,6 +116,26 @@ export class AmpForm {
     }
     this.form_.classList.add(`amp-form-${state}`);
   }
+
+  /**
+   * @param {string} state
+   * @param {!Object} data
+   * @private
+   */
+  renderTemplate_(state, data) {
+    const container = this.form_.querySelector(`[${state}]`);
+    if (container) {
+      const previousRender = container.querySelector(`.${state}-message`);
```

Comment:
Mind if I do this in a separate PR. The other PR I have open introduce the `this.state_` property and don't wanna introduce it again in this one.